### ISSUE TITLE
Make Phone call directly from selecting contact > number

### ISSUE THREAD

**Issue by fossimprove** [MM2-0/Kvaesitso#163]
Firstly, One of the best launchers I have used. Thanks a lot, MM2-0!!!!
This brought me proper use for widgets.

Now coming to issue : 
For search, the permission for contacts should serve both Call and SMS.
Please try to make Phone call directly from clicking on number from contact box (search) 
You can make it option to toggle in setting,

The default behavior is, it puts that number in dialer (It takes more than 3 steps, which is not required)
Also add button to message that contact directly, even if we have not started messaging with that contact.

Or else, this feature becomes no use for searching.

For reference, KISS launcher does it. It does by simply showing call or message icon beside contact. That is enough and serves the purpose.

I hope you get my point, let me know if any clarification or if you didn't get me.

Thank you very much MM2-0!

**Comment by MM2-0** [comment]
Honestly, I'm not a huge fan of this because I'm too scared of calling someone by accident. But if it's optional… maybe in the future.

**Comment by fossimprove** [comment]
I mean, since we select that contact box (in search), we are conscious of who we are calling.3
We can click on this to directly make call. 

  [Call from clicking here](https://0x0.st/oVyi.jpg)

**Comment by tzugen** [comment]
I came also here to open an Feature request about this.
Otherwise, thanks @MM2-0  I am really enjoying this launcher

**Comment by nv2p** [comment]
I would also like this to be implemented. I agree with fossimprove and when I'm in hurry, your system doesn't prevent mistakes (like mixing people; yeah, it happened...). I don't find it so horrible to call the wrong person (shit happens) and on the other hand, I appreciate very much the few seconds saved when I quickly want to call someone, doing many things at once...

Until you consider a 1 step system like KISS (which would mean changing your sorting when a contact have more than 1 number, I think...) would it be possible to at least reduce from 3 to 2 steps, by calling straight when the phone number is pressed?

I would also strongly suggest the feature of opening the contact in the contacts app when the name is pressed in your "menu" (after I clicked on it in the search), instead of the (annoying) extra step of pressing the 3 dots first (or you could put the symbol next to it and the star which save it in the favorites).

By the way (not so important for me, but I can open an issue for this if you want), I don't have the text message (SMS) possibility in your contact "menu" and I have it marked in the quick actions.

**Comment by exstntlmsnthrp** [comment]
@nv2p I agree with the points that you make. I'd not thought about it in this way, but it makes sense. I like the idea of opening in whatever contacts app you have or phoning the person. 

It would be nice to see that implemented.

**Comment by tzugen** [comment]
I think there could be a nicer way even:
In the native contact app, there is a number of buttons, which allow me to Call or message a contact through the native methods (call/sms) or through messenger apps. 
I have not understood yet which API allows one to enumerate this list of contact possibilities.
The source code is here: https://android.googlesource.com/platform/packages/apps/Contacts/

**Comment by tzugen** [comment]
Ah, here it is:
https://stackoverflow.com/questions/46314462/how-to-get-all-social-accounts-linked-with-the-particular-contact-in-android-fro

**Comment by SysadminLink** [comment]
#347

**Comment by trbhlz** [comment]
I came here to make the same suggestion but now i'm disappointed because ist hasn't been realized 2 years after suggestion. In many  years of using kiss launcher it never happend to that i have called the wrong Person. It's so much smoother to call my contacts directly and is the main feature for me that makes kiss launcher so special. I like your launcher very much and in my opinion it's much better than kiss, but because of the lack of this feature and the fact that i have to scroll to find the contacts i go back to kiss. Please make the direct call  button, i would rather use your launcher

**Comment by toas-koas** [comment]
Is there an update for this? 
I think it's great when the contact is called while typing.
